### ISSUE TITLE
huawei support causes crash when calling "stopLocationUpdates"

### ISSUE THREAD

**Issue by mrzbn** [BirjuVachhani/locus-android#90]
probably adding HMS dependency solves the problem, but is there a way that I'm not forced to add the dependency?

**Comment by ruifernando7** [comment]
any update on this? i'm having the same issue. The app crashed in runtime @BirjuVachhani

**Comment by BirjuVachhani** [comment]
I haven't gotten around it yet! If I understand this correctly, calling `startLocationUpdates` (without adding HMS dependency) works, only calling `stopLocationUpdates` fails, right?

**Comment by Tushar-Salvi** [comment]
yes, @BirjuVachhani calling stopLocationUpdates on pause or in background crash the app, here the code where it is crashing i.e LocationProvider class 

internal fun stopUpdates() {
        logDebug("Stopping background location updates")
        isRequestOngoing.set(false)
        locationLiveData = MutableLiveData()
        gmsFusedLocationProviderClient.removeLocationUpdates(pendingIntent)
        hmsFusedLocationProviderClient.removeLocationUpdates(pendingIntent) // crash here
    }
